### ISSUE TITLE
No testing for cl_khr_il_program

### ISSUE THREAD

**Issue by kpet** [KhronosGroup/OpenCL-CTS#479]
We should re-use the OpenCL 2.1 tests. It seems we'd need to do at least the following:
- [ ] Exercise `CL_DEVICE_IL_VERSION_KHR`
- [ ] Test `CL_PROGRAM_IL_KHR` (see #164)
- [ ] Use the extension entrypoint in the harness and existing tests 
- [ ] Decide what to do on implementations that support both the core feature and the extension

**Comment by alycm** [comment]
I assume that some of the work for this was done in the "cl12: Backport spirv_new from cl21_trunk to test cl_khr_il_program" PR: https://github.com/KhronosGroup/OpenCL-CTS/pull/219

Just in the wrong direction for the purposes of this issue.

**Comment by kpet** [comment]
Yes, indeed. Thanks for linking the PR. I actually got it going on the master branch with some similar-looking changes. I'll close this PR now that it's linked here since we decided to enable 1.2 on master instead and we need to be able to test the core feature and extension from the same binary.

**Comment by gwawiork** [comment]
Can somebody confirm that this spirv_new changes should be merged to master but looks like we lost it ?

**Comment by rjodinchr** [comment]
Fixed: `testHarness.cpp` and `spirv_new/test_get_program_il.cpp` execute full SPIR-V conformance tests on OpenCL < 2.1 devices when the `cl_khr_il_program` extension is advertised.
